devops/kubernetes/storage-node/index.ts

```diff
@@ -96,74 +89,22 @@ const pvc = new k8s.core.v1.PersistentVolumeClaim(
   resourceOptions
 )
 
-volumes.push({
-  name: 'ipfs-data',
-  persistentVolumeClaim: {
-    claimName: `${name}-pvc`,
+const keyFile = config.require('keyFile')
+const keyConfigName = new configMapFromFile(
+  'key-config',
+  {
+    filePath: keyFile,
+    namespaceName: namespaceName,
   },
-})
-
-const caddyEndpoints = [
-  ` {
-    reverse_proxy storage-node:${colossusPort}
-}`,
-]
-
-export let endpoint1: pulumi.Output<string> = pulumi.interpolate``
-export let endpoint2: pulumi.Output<string> = pulumi.interpolate``
-
-if (!isMinikube) {
-  const caddy = new CaddyServiceDeployment(
-    'caddy-proxy',
-    { lbReady, namespaceName: namespaceName, caddyEndpoints },
-    resourceOptions
-  )
-
-  endpoint1 = pulumi.interpolate`${caddy.primaryEndpoint}`
-  endpoint2 = pulumi.interpolate`${caddy.secondaryEndpoint}`
-}
-
-export let appLink: pulumi.Output<string>
-
-if (lbReady) {
-  appLink = pulumi.interpolate`https://${endpoint1}`
-
-  if (!isAnonymous) {
-    const remoteKeyFilePath = '/joystream/key-file.json'
-    const providerId = config.require('providerId')
-    const keyFile = config.require('keyFile')
-    const publicUrl = config.get('publicURL') ? config.get('publicURL')! : appLink
-
-    const keyConfig = new k8s.core.v1.ConfigMap('key-config', {
-      metadata: { namespace: namespaceName, labels: appLabels },
-      data: { 'fileData': fs.readFileSync(keyFile).toString() },
-    })
-    const keyConfigName = keyConfig.metadata.apply((m) => m.name)
-
-    additionalParams = ['--provider-id', providerId, '--key-file', remoteKeyFilePath, '--public-url', publicUrl]
-
-    volumeMounts.push({
-      mountPath: remoteKeyFilePath,
-      name: 'keyfile-volume',
-      subPath: 'fileData',
-    })
+  resourceOptions
+).configName
 
-    volumes.push({
-      name: 'keyfile-volume',
-      configMap: {
-        name: keyConfigName,
-      },
-    })
+const remoteKeyFilePath = '/joystream/key-file.json'
+additionalParams = ['--key-file', remoteKeyFilePath]
```

Comment:
keyfile is required above but additionalParams are not used, since we are passing ACCOUNT_URI.

So we need better handling I think here of keyFile, passphrase, and accountUri configs, especially if both are set, which one takes precedence (based on storage-node implementation of course)

Reply:
So according to the method `getAccount` in `storage-node-v2/src/command-base/ApiCommandBase.ts` if `accountURI` is specified it takes precedence and overrides `password` and `keyFile` flags.

Also either `Keyfile` or `accountURI` must be set otherwise we throw an error.

Will update this in the code.